client/mm/mm.go

```diff
@@ -219,6 +285,55 @@ func priceOracleFromConfigs(ctx context.Context, cfgs []*BotConfig, log dex.Logg
 	return oracle, nil
 }
 
+func (m *MarketMaker) markBotAsRunning(id string, running bool) {
+	m.runningBotsMtx.Lock()
+	defer m.runningBotsMtx.Unlock()
+	if running {
+		m.runningBots[id] = struct{}{}
+	} else {
+		delete(m.runningBots, id)
+	}
+
+	if len(m.runningBots) == 0 {
+		m.die()
+	}
+}
+
+// MarketReport returns information about the oracle rates on a market
+// pair and the fiat rates of the base and quote assets.
+func (m *MarketMaker) MarketReport(base, quote uint32) (*MarketReport, error) {
+	user := m.core.User()
+	baseFiatRate := user.FiatRates[base]
+	quoteFiatRate := user.FiatRates[quote]
+
+	m.syncedOracleMtx.RLock()
+	if m.syncedOracle != nil {
+		price, oracles, err := m.syncedOracle.GetOracleInfo(base, quote)
+		m.syncedOracleMtx.RUnlock()
+		if err == nil {
```

Comment:
It may not be an error, it may just be that the SyncedOracle does not have a price for that market. I can return a specific error for this and then log if it is a different error.